tests/test_coordinates.py

```diff
@@ -189,3 +190,34 @@ def test_equatorial_to_eci_coords():
         _ra, _dec = coordinates.equatorial_to_eci_coords(
             ra, dec, obstime, (-30.7, 21.4, 1073.0), unit="rad", frame="icrs"
         )
+
+
+def test_coherency_calc():
+    """Test calculation of coherency matrix."""
+    params, sky_model, *_ = get_standard_sim_params(
+        use_analytic_beam=False,
+        polarized=True,
+        nsource=NPTSRC,
+        ntime=NTIMES,
+    )
```

Comment:
In addition to some random sources, it might be a good idea to test "special" cases, like a point at zenith, anti-zenith, and on the horizon at EW and NS.